tests/external_aiobotocore/test_aiobotocore_sqs.py

```diff
@@ -0,0 +1,113 @@
+from testing_support.validators.validate_transaction_metrics import (
+    validate_transaction_metrics,
+)
+
+from newrelic.api.background_task import background_task
+
+URL = "localhost:%s" % PORT
+TEST_QUEUE = "python-agent-test"
+
+_sqs_scoped_metrics = [
+    ("MessageBroker/SQS/Queue/Produce/Named/%s" % TEST_QUEUE, 2),
+    ("External/%s/aiobotocore/POST" % URL, 6),
+]
+
+_sqs_rollup_metrics = [
+    ("MessageBroker/SQS/Queue/Produce/Named/%s" % TEST_QUEUE, 2),
+    # ("MessageBroker/SQS/Queue/Consume/Named/%s" % TEST_QUEUE, 1),
+    ("External/all", 6),
+    ("External/allOther", 6),
+    ("External/%s/all" % URL, 6),
+    ("External/%s/aiobotocore/POST" % URL, 6),
+]
+
+
+@validate_span_events(exact_agents={"aws.operation": "CreateQueue"}, count=1)
+@validate_span_events(exact_agents={"aws.operation": "ListQueues"}, count=1)
+@validate_span_events(exact_agents={"aws.operation": "SendMessage"}, count=1)
+# @validate_span_events(exact_agents={"aws.operation": "ReceiveMessage"}, count=1)
+@validate_span_events(exact_agents={"aws.operation": "SendMessageBatch"}, count=1)
+@validate_span_events(exact_agents={"aws.operation": "PurgeQueue"}, count=1)
+@validate_span_events(exact_agents={"aws.operation": "DeleteQueue"}, count=1)
+@validate_transaction_metrics(
+    "test_aiobotocore_sqs:test_aiobotocore_sqs",
+    scoped_metrics=_sqs_scoped_metrics,
+    rollup_metrics=_sqs_rollup_metrics,
+    background_task=True,
+)
+@background_task()
+def test_aiobotocore_sqs(loop):
+    async def _test():
+        async with MotoService("sqs", port=PORT):
+            session = get_session()
+
+            async with session.create_client(
+                "sqs",
+                region_name="us-east-1",
+                endpoint_url="http://localhost:%d" % PORT,
+                aws_access_key_id=AWS_ACCESS_KEY_ID,
+                aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
+            ) as client:
+
+                response = await client.create_queue(QueueName=TEST_QUEUE)
+
+                queue_url = response["QueueUrl"]
+
+                # List queues
+                response = await client.list_queues()
+                for queue_name in response.get("QueueUrls", []):
+                    assert queue_name
+
+                # Send message
+                resp = await client.send_message(QueueUrl=queue_url, MessageBody="hello_world")
+                assert resp["ResponseMetadata"]["HTTPStatusCode"] == 200
+
+                # Receive message
+                # This works in an actual application, but not with the proxy set up
+                # for the test.  The aiobotocore's proxy requests redirect to aiohttp's
+                # request call which results in double instrumentation and invalid XML
+                # generation in response_dict["headers"]["body"]
+                # resp = await client.receive_message(QueueUrl=queue_url)
+                # assert resp["ResponseMetadata"]["HTTPStatusCode"] == 200
```

Comment:
Are we sure this actually works with a real AWS connection running through a proxy? Or does the existence of a proxy cause issues here?

Reply:
In this case, the aiobotocore's proxy functionality uses aiohttp and urllib3 under the hood.  Because it used those under the hood, New Relic's browser monitoring was injecting `_js_agent_header_fragment` into the response.  This in turn caused the response message to be returned as an invalid XML.